awslabs/eventbridge-kafka-connector
# Add `JSONPathDetailTypeMapper`

Description
-----------

Introduces `JsonPathDetailTypeMapper` to enable dynamic extraction of EventBridge detail-type values from Kafka messages using JSONPath expressions. The mapper evaluates the configured JSONPath expression against the Kafka message payload and uses the extracted value as the detail-type in the resulting EventBridge event. This enhancement supports more complex message structures and enables better event routing capabilities in EventBridge based on the content of Kafka messages.

Test Steps
-----------

Implemented unit tests for `JsonPathDetailTypeMapper` and `DefaultEventBridgeMapper` to verify the behavior.

Checklist:
----------




Related Issue
-----------

#272 


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
